recipes/ncbi-cxx-toolkit-public/26.0/conanfile.py

```diff
@@ -0,0 +1,1450 @@
+                return None
+            if key == "CASSANDRA" and (self.settings.os == "Windows" or self.settings.os == "Macos"):
+                return None
+            if key == "NGHTTP2" and self.settings.os == "Windows":
+                return None
+            return self.NCBI_to_Conan_requires[key]
+        return None
+
+#----------------------------------------------------------------------------
+    @property
+    def _source_subfolder(self):
+        return "src"
+
+#----------------------------------------------------------------------------
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        cmake.definitions["NCBI_PTBCFG_PACKAGING"] = "TRUE"
+        if self.options.with_projects != "":
+            cmake.definitions["NCBI_PTBCFG_PROJECT_LIST"] = self.options.with_projects
+        if self.options.with_targets != "":
+            cmake.definitions["NCBI_PTBCFG_PROJECT_TARGETS"] = self.options.with_targets
+        return cmake
+
+#----------------------------------------------------------------------------
+    def validate(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, 17)
+        if self.settings.os not in ["Linux", "Macos", "Windows"]:   
+            raise ConanInvalidConfiguration("This operating system is not supported")
+        if self.settings.compiler == "Visual Studio" and str(self.settings.compiler.version) < "15":
+            raise ConanInvalidConfiguration("This version of Visual Studio is not supported")
+        if self.settings.compiler == "Visual Studio" and self.options.shared and "MT" in self.settings.compiler.runtime:
+            raise ConanInvalidConfiguration("This configuration is not supported")
+        if self.settings.compiler == "gcc" and str(self.settings.compiler.version) < "7":
+            raise ConanInvalidConfiguration("This version of GCC is not supported")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+#----------------------------------------------------------------------------
+    def requirements(self):
+        NCBIreqs = self._get_RequiresMapKeys()
+        for req in NCBIreqs:
+            pkg = self._translate_ReqKey(req)
+            if pkg is not None:
+                self.requires(pkg)
+
+#----------------------------------------------------------------------------
+    def source(self):
+        tk_url = self.conan_data["sources"][self.version]["url"]
+        print("getting Toolkit sources...")
+        print("from url: " + tk_url)
+        tools.get(**self.conan_data["sources"][self.version], strip_root = True)
+
+#----------------------------------------------------------------------------
+    def imports(self):
+        self.copy("license*", dst="licenses", folder=True, ignore_case=True)
```

Comment:
```suggestion
```

useless

Reply:
I follow this recommendation
https://docs.conan.io/en/latest/howtos/collect_licenses.html
what do you mean "useless"??

Reply:
It's useless because `imports()` copy files to build folder, so we don't care here. What you want is to copy licenses of NCBI C++ toolkit (and only this library, not its dependencies) in package folder, and it's done in `package()`.

Reply:
My experience with licenses stuff is that it is better to have more than less. useless? - ok, at least not harmful

Reply:
1. licenses of dependencies are already available in their own packages.
2. `imports()` doesn't copy files in package folders anyway.

A package is not more, not less, just exactly what is needed.